src/api/ledger/parse/utils.js

```diff
@@ -4,6 +4,15 @@ const _ = require('lodash');
 const transactionParser = require('ripple-lib-transactionparser');
 const toTimestamp = require('../../../core/utils').toTimestamp;
 const utils = require('../utils');
+const BigNumber = require('bignumber.js');
+
+function adjustQualityForXRP(quality: string, takerGetsCurrency: string,
+    takerPaysCurrency: string) {
+  const shift = (takerGetsCurrency === 'XRP' ? 6 : 0)
```

Comment:
It's correct at least now :) Arguably it was only incorrect before due to this convoluted construct. Do we have some hard and fast rule to avoid nested ternary expressions or something?